peerconnection.go

```diff
@@ -510,7 +510,13 @@ func (pc *PeerConnection) onConnectionStateChange(cs PeerConnectionState) {
 	pc.connectionState.Store(cs)
 	pc.log.Infof("peer connection state changed: %s", cs)
 	if handler, ok := pc.onConnectionStateChangeHandler.Load().(func(PeerConnectionState)); ok && handler != nil {
-		go handler(cs)
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			handler(cs)
+			wg.Done()
+		}()
+		wg.Wait()
```

Comment:
Why wg instead of just removing `go`?

Reply:
@Wondertan this is so the callback doesn’t block the ICE Agent

hopefully we have a test for that :)

Reply:
I believe we have this issue with this PR. The code right now would allow a callback to block the PeerConnection internals.

Reply:
>Why wg instead of just removing go?

Yeah sorry this is not required at all.

Reply:
No worries @sukunrt, all good